lib/Transforms/ElementwiseToAffine/ElementwiseToAffine.cpp

```diff
@@ -0,0 +1,136 @@
+#include "include/Transforms/ElementwiseToAffine/ElementwiseToAffine.h"
+
+#include "mlir/include/mlir/Dialect/Affine/IR/AffineOps.h"  // from @llvm - project
+#include "mlir/include/mlir/Dialect/Func/IR/FuncOps.h"   // from @llvm-project
+#include "mlir/include/mlir/Dialect/Tensor/IR/Tensor.h"  // from @llvm-project
+#include "mlir/include/mlir/Transforms/DialectConversion.h"  // from @llvm-project
+#include "mlir/include/mlir/Transforms/Passes.h"  // from @llvm-project
+namespace mlir {
+namespace heir {
+
+#define GEN_PASS_DEF_ELEMENTWISETOAFFINE
+#include "include/Transforms/ElementwiseToAffine/ElementwiseToAffine.h.inc"
+
+// All of this is based on the ElementwiseToLinalg Pass in
+// mlir/lib/Dialect/Linalg/Transforms/ElementwiseToLinalg.cpp
+
+static bool isElementwiseMappableOpOnRankedTensors(Operation *op) {
+  if (!OpTrait::hasElementwiseMappableTraits(op)) return false;
+
+  // FIXME: Test this with `any_of`,
```

Comment:
If you prefer not to fix this in this PR, file an issue and link it to this line with

```
// TODO(#<issue_number>): message that will show up in the GH issue
```

I wrote https://github.com/j2kun/todo-backlinks to automate managing these over the lifecycle of the issue.